netmaster/daemon/daemon.go

```diff
@@ -404,7 +409,11 @@ func (d *MasterDaemon) runLeader() {
 	router := mux.NewRouter()
 
 	// Create a new api controller
-	d.apiController = objApi.NewAPIController(router, d.objdbClient, d.ClusterStore)
+	apiConfig := &objApi.APIControllerConfig{
+		NetForwardMode: d.NetForwardMode,
+		NetInfraType:   d.NetInfraType,
+	}
```

Comment:
this object only got used here (and integ testings), I used object was because of intending to add more parameters from configs . 
> have NewAPIController accept APIControllerConfig by value
this is troublesome in golang, especially when having the parameters in different type

Reply:
i think the struct is fine, just dunno why you need a pointer to it